pkg/codegen/pcl/binder_test.go

```diff
@@ -749,3 +749,84 @@ value = element(data, 0)
 	assert.Empty(t, diags, "There are no error or warning diagnostics")
 	assert.NotNil(t, program)
 }
+
+func TestBindingSelfReferencingComponentFailsWithCircularReference(t *testing.T) {
```

Comment:
nit: WithSelfReferenceWhenCircular

or something to that effect?

Reply:
"SelfReferencing" already implies a circular reference with self :thinking: I read the test like this:
"Test binding self-referencing components fails with a circular reference error" 

I can add a comment about the test if it is not clear enough

Reply:
Added a comment explaining what the test does ✅